## Loan prediction

In [88]:
## First import basic pakages

import numpy as np
import pandas as pd

In [89]:
ldata= pd.read_csv("D:\\4 Aug 2020\\Loan prediction\\loan train data.csv")

### Data introduction

In [90]:
ldata.head(5)

Loan_ID Gender Married Dependents     Education Self_Employed  \
0  LP001002   Male      No          0      Graduate            No   
1  LP001003   Male     Yes          1      Graduate            No   
2  LP001005   Male     Yes          0      Graduate           Yes   
3  LP001006   Male     Yes          0  Not Graduate            No   
4  LP001008   Male      No          0      Graduate            No   

   ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0             5849                0.0         NaN             360.0   
1             4583             1508.0       128.0             360.0   
2             3000                0.0        66.0             360.0   
3             2583             2358.0       120.0             360.0   
4             6000                0.0       141.0             360.0   

   Credit_History Property_Area Loan_Status  
0             1.0         Urban           Y  
1             1.0         Rural           N  
2             1.0         Urban           Y  
3             1.0         Urban           Y  
4             1.0         Urban           Y

In [91]:
ldata.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            614 non-null    object 
 1   Gender             601 non-null    object 
 2   Married            611 non-null    object 
 3   Dependents         599 non-null    object 
 4   Education          614 non-null    object 
 5   Self_Employed      582 non-null    object 
 6   ApplicantIncome    614 non-null    int64  
 7   CoapplicantIncome  614 non-null    float64
 8   LoanAmount         592 non-null    float64
 9   Loan_Amount_Term   600 non-null    float64
 10  Credit_History     564 non-null    float64
 11  Property_Area      614 non-null    object 
 12  Loan_Status        614 non-null    object 
dtypes: float64(4), int64(1), object(8)
memory usage: 62.5+ KB


This loan data is used to predict which customer will get home loan or approve thier home loan.in this loan data contains customers Basic inforation like gender, marrital status, education...


in loan data contains 614 observations(rows) and 13 features (columns).

types of features

catogory type - 8 columns (features)
numerical type - 5 columns (features)

Our target variable is Loan_status because from given data we actually predict Loan_status.




#### Statistics

In [92]:
ldata.describe()

ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
count       614.000000         614.000000  592.000000         600.00000   
mean       5403.459283        1621.245798  146.412162         342.00000   
std        6109.041673        2926.248369   85.587325          65.12041   
min         150.000000           0.000000    9.000000          12.00000   
25%        2877.500000           0.000000  100.000000         360.00000   
50%        3812.500000        1188.500000  128.000000         360.00000   
75%        5795.000000        2297.250000  168.000000         360.00000   
max       81000.000000       41667.000000  700.000000         480.00000   

       Credit_History  
count      564.000000  
mean         0.842199  
std          0.364878  
min          0.000000  
25%          1.000000  
50%          1.000000  
75%          1.000000  
max          1.000000

### Data cleaning

In [93]:
ldata['Credit_History'].unique()

array([ 1.,  0., nan])

#### 'credit history' is categorical variable but here machine is consider its numerical so we need to change the type of this variable

In [94]:
ldata['Credit_History']=ldata['Credit_History'].astype(object)

In [95]:
ldata.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            614 non-null    object 
 1   Gender             601 non-null    object 
 2   Married            611 non-null    object 
 3   Dependents         599 non-null    object 
 4   Education          614 non-null    object 
 5   Self_Employed      582 non-null    object 
 6   ApplicantIncome    614 non-null    int64  
 7   CoapplicantIncome  614 non-null    float64
 8   LoanAmount         592 non-null    float64
 9   Loan_Amount_Term   600 non-null    float64
 10  Credit_History     564 non-null    object 
 11  Property_Area      614 non-null    object 
 12  Loan_Status        614 non-null    object 
dtypes: float64(3), int64(1), object(9)
memory usage: 62.5+ KB


In [96]:
#### check the missing values 
ldata.isnull().sum()

Loan_ID               0
Gender               13
Married               3
Dependents           15
Education             0
Self_Employed        32
ApplicantIncome       0
CoapplicantIncome     0
LoanAmount           22
Loan_Amount_Term     14
Credit_History       50
Property_Area         0
Loan_Status           0
dtype: int64

here we required to fill the missing values 

In [97]:
ldata['Credit_History'].value_counts()

1.0    475
0.0     89
Name: Credit_History, dtype: int64

In [98]:
ldata.loc[pd.isnull(ldata['Credit_History']), ['Credit_History']] = 1

In [99]:
ldata['Gender'].value_counts()

Male      489
Female    112
Name: Gender, dtype: int64

In [100]:
ldata.loc[pd.isnull(ldata['Gender']), ['Gender']] = 'Male'

In [101]:
ldata['Dependents'].value_counts()

0     345
1     102
2     101
3+     51
Name: Dependents, dtype: int64

In [102]:
ldata.loc[pd.isnull(ldata['Dependents']), ['Dependents']] = 'Male'

In [103]:
ldata['Married'].value_counts()

Yes    398
No     213
Name: Married, dtype: int64

In [104]:
ldata.loc[pd.isnull(ldata['Married']), ['Married']] = 'Yes'

In [105]:
ldata['Self_Employed'].value_counts()

No     500
Yes     82
Name: Self_Employed, dtype: int64

In [106]:
ldata.loc[pd.isnull(ldata['Self_Employed']), ['Self_Employed']] = 'No'

In [107]:
ldata['LoanAmount'].describe()

count    592.000000
mean     146.412162
std       85.587325
min        9.000000
25%      100.000000
50%      128.000000
75%      168.000000
max      700.000000
Name: LoanAmount, dtype: float64

In [108]:
ldata.loc[pd.isnull(ldata['LoanAmount']), ['LoanAmount']] = 128

In [109]:
ldata['Loan_Amount_Term'].describe()

count    600.00000
mean     342.00000
std       65.12041
min       12.00000
25%      360.00000
50%      360.00000
75%      360.00000
max      480.00000
Name: Loan_Amount_Term, dtype: float64

In [110]:
ldata.loc[pd.isnull(ldata['Loan_Amount_Term']), ['Loan_Amount_Term']] = 360

In [111]:
ldata.isnull().sum()

Loan_ID              0
Gender               0
Married              0
Dependents           0
Education            0
Self_Employed        0
ApplicantIncome      0
CoapplicantIncome    0
LoanAmount           0
Loan_Amount_Term     0
Credit_History       0
Property_Area        0
Loan_Status          0
dtype: int64

In [113]:
ldata.drop(['Loan_ID'],inplace=True,axis=1)

In [114]:
ldata.to_csv('cleaned data.csv')